contracts/utils/Checkpoints.sol

```diff
@@ -68,6 +68,14 @@ library Checkpoints {
         return pos == 0 ? 0 : _unsafeAccess(self._checkpoints, pos - 1)._value;
     }
 
+    /**
+     * @dev Returns checkpoint at given position.
+     */
+    function getAtPosition(History storage self, uint32 pos) internal view returns (Checkpoint memory) {
+        Checkpoint storage last = _unsafeAccess(self._checkpoints, pos);
+        return last;
```

Comment:
I don't think we should use `_unsafeAccess` here, we should be checking that the position is within bounds.

```suggestion
        return self._checkpoints[pos];
```